src/applet.c

```diff
@@ -227,8 +231,16 @@ static void brisk_menu_applet_init(BriskMenuApplet *self)
         gtk_widget_set_margin_end(image, 4);
         gtk_widget_set_halign(image, GTK_ALIGN_START);
 
+        /* Set the label text, if empty, show the default text */
+        gchar *text = g_settings_get_string(self->settings, "label-text");
```

Comment:
Leak. Also this should likely happen during pump settings or you just fire off  the change callback